### DISCUSSION TITLE
How to draw polygons in Klein model

### DISCUSSION THREAD

**Discussion by neozhaoliang** [cduck/hyperbolic#16] [Q&A] [answered]
Is it possible to draw polygons in Klein model?  In the unit disk a polygon has straight edges and I want to fill the inside with color.

**Comment by cduck** [comment, marked as answer]
Thanks for your question.  The Klein model is not currently supported.

To support the Klein model, you could implement a new kind of `Transform` similar to `Transform.diskToHalf()` used in the [half-plane example](https://github.com/cduck/hyperbolic/blob/master/examples/poincare.ipynb).  Map the Poincare coordinates to Klein using [these equations](https://en.wikipedia.org/wiki/Poincar%C3%A9_disk_model#Relation_to_the_Klein_disk_model).

**Reply by neozhaoliang** [reply]
Thanks!
